src/main.rs

```diff
@@ -230,20 +235,26 @@ impl From<PingEvent> for Event {
 #[behaviour(event_process = false)]
 #[behaviour(out_event = "Event")]
 struct Behaviour {
-    ping: Ping,
+    ping: Toggle<Ping>,
     rendezvous: Rendezvous,
 }
 
 impl Behaviour {
-    fn new(rendezvous: Rendezvous) -> Self {
-        Self {
-            // TODO: Remove Ping behaviour once https://github.com/libp2p/rust-libp2p/issues/2109 is fixed
-            // interval for sending Ping set to 24 hours
-            ping: Ping::new(
+    fn new(rendezvous: Rendezvous, ping: bool) -> Self {
+        let ping = if ping {
+            Toggle::from(Some(Ping::new(
                 PingConfig::new()
                     .with_keep_alive(false)
                     .with_interval(Duration::from_secs(86_400)),
-            ),
+            )))
+        } else {
+            Toggle::from(None)
+        };
```

Comment:
Rust stdlib recently got [`bool::then`](https://doc.rust-lang.org/stable/std/primitive.bool.html#method.then).

Allows you to specify a closure that only gets run if the bool is `true`, producing an `Option` in the end. With that, you could write this as:

```suggestion
let ping = Toggle::from(ping.then(|| Ping::new(PingConfig::new(...)));
```

If you rename the parameter to be more "bool"-ish, i.e. express it as a _state_ like `enable_ping`, it might be even more expressive:

```rust
let ping = Toggle::from(enable_ping.then(|| Ping::default()));
```